examples/addsvc/client/grpc/client.go

```diff
@@ -36,27 +48,41 @@ type client struct {
 //   important.
 
 func (c client) Sum(a, b int) int {
-	request := &pb.SumRequest{
-		A: int64(a),
-		B: int64(b),
+	request := &server.SumRequest{
+		A: a,
+		B: b,
 	}
-	reply, err := c.AddClient.Sum(c.Context, request)
+	reply, err := c.sumEndpoint(c.Context, request)
 	if err != nil {
 		c.Logger.Log("err", err) // Without an error return parameter, we can't do anything else...
 		return 0
 	}
-	return int(reply.V)
+
+	r, ok := reply.(server.SumResponse)
+	if !ok {
```

Comment:
I think if this is not true, it's a programmer error, and it's better to just panic. So this (and all other type assertions) can elide the `, ok` check, IMO.